### PR TITLE
(TESTING) Pull Request for OS Only pi-factory

### LINKED ISSUE #236: OS Only version of pi-factory
Use the latest version of Alpine. 

Remove all the bitcoind / lnd node cruft from this and simplify it. 

Have the image script build just an alpine OS image which can easily install the ```noma``` package OR use for some other purpose.

## Requirements

- [x] Support for ```aarch64```. The raspberry pi 3 is more than capable of handling this
- [x] Keep support for ```armhf``` for the zero

## Issues

- [x] IOTWIFI does not work with the latest version of Alpine, however there are workarounds for this.

### PR DESCRIPTION
**Pull request by nolim1t**
Pull request for OS Only pi-factory (When merged this should close #236 )

## Contains
- [x]  Latest version alpine
- [x]  Modified wpa_supplicant script to detect for wpa_supplicant.conf file in FAT partition
- [x]  Latest version alpine cache
- [x] Pull request #237 
- [x] Github Action

## TODO

- [x] Test out Raspberry PI 3/3b
- [x] Create Raspberry Pi Zero alternate config
- [x] Test Raspberry Pi Zero alternate config

## Known Issues

- [x] Install fails if no internet, but this is handled. How to handle this in a userfriendly manner?

## Testing

```bash
git clone https://github.com/nolim1t/pi-factory -b issue-236 pi-factory-os-only
cd pi-factory-os-only
sudo ./make_img.sh # or on mac 'vagrant up'
```

### CODE REVIEW

File: etc/init.d/lncm-online [hunk 1]
```diff
@@ -18,17 +18,21 @@ start() {
...
+  while [ $ONLINE -eq 1 ] && [ $COUNT -lt 10 ]
   do
-      echo "Checking for internet connection..."
+      echo "Checking for internet connection for 10 seconds..."
       net_check
+      COUNT=$((COUNT + 1))
       sleep 1
   done
 
   if [ $ONLINE -eq 0 ]; then
```
Comment (meeDamian) [inline etc/init.d/lncm-online]: `$ONLINE` is never changed

File: etc/init.d/lncm-online [hunk 2]
```diff
@@ -18,17 +18,21 @@ start() {
...
     echo "Connected to internet"
     exit 0
+  else
```
Comment (meeDamian) [inline etc/init.d/lncm-online]: `else` unnecessary, as `if` does `exit 0`.

And maybe better to move that check into the loop? Something like:

```sh
COUNT=0
while [ "${COUNT}" -lt "10" ]; do
	printf "Checking internet connection #%d…\t" "$(COUNT)"

	echo -e 'GET http://dl-cdn.alpinelinux.org/alpine/ HTTP/1.0\n\n' | \
		timeout 5s nc dl-cdn.alpinelinux.org 80 > /dev/null 2>&1

	if [ "$?" -ne "0" ]; then
		echo "OK!"
		exit 0
	fi
	
	echo "ERR"
	COUNT=$((COUNT + 1))
	sleep 1
done

echo "Not connected to internet after 30 seconds"
```

File: etc/init.d/lncm-post [hunk 3]
```diff
@@ -29,6 +29,19 @@ start() {
     fi
     /sbin/setup-apkcache /var/cache/apk
   }
+  
+  install_python3() {
+    echo "Install Python 3 and upgrade PIP"
+    /sbin/apk update && \
+    /sbin/apk add python3 python3-dev curl jq
```
Comment (meeDamian) [inline etc/init.d/lncm-post]: Perhaps `curl`, and `jq` should be in sth like `install_utils` fn?

File: etc/init.d/lncm-post [hunk 4]
```diff
@@ -267,8 +280,8 @@ start() {
   }
 
   persist_state() {
-    echo "Backup apkovl.tar.gz to USB storage"
-    /sbin/lbu package /media/important/important/box.apkovl.tar.gz
+    echo "Backup apkovl.tar.gz to Root"
```
Comment (meeDamian) [inline etc/init.d/lncm-post]: "…to filesystem root (`/`)".

File: etc/ssh/sshd_config [hunk 5]
```diff
@@ -19,102 +7,22 @@
 #HostKey /etc/ssh/ssh_host_ecdsa_key
 #HostKey /etc/ssh/ssh_host_ed25519_key
 
-# Only generate ed25519 and RSA host keys
-HostKeyAlgorithms ssh-ed25519-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com,ssh-ed25519,ssh-rsa
```
Comment (meeDamian) [inline etc/ssh/sshd_config]: Please leave this in!

File: etc/wpa_supplicant/wpa_supplicant.conf [hunk 6]
```diff
@@ -1,10 +1,5 @@
-# Please change to match your country code
-country=US
-update_config=1
-ctrl_interface=/var/run/wpa_supplicant
```
Comment (meeDamian) [inline etc/wpa_supplicant/wpa_supplicant.conf]: This is necessary on Raspbian, are you sure it's not on Alpine?

File: etc/apk/world [hunk 7]
```diff
@@ -1,7 +1,11 @@
 alpine-base
-avahi
 dbus
+avahi
+chrony
 docker
```
[the remark this answers is not in the document]
Comment (nolim1t) [reply etc/apk/world]: makes it easier to get up noma if they opt in to it.

[the remark this answers is not in the document]
Comment (meeDamian) [reply etc/apk/world]: Perhaps we should have `-barebone`, and `-recommended` versions?

File: etc/apk/world [hunk 8]
```diff
@@ -1,7 +1,11 @@
...
 openssh
+openssl
 wireless-tools
 wpa_supplicant
+e2fsprogs
+parted
```
[the remark this answers is not in the document]
Comment (nolim1t) [reply etc/apk/world]: Most of these were from the base image after running setup-alpine.

parted, and e2fsprogs is to partition the device.

Openssl I think is a dependency. Avahi is a nice addition for those who don't have an external display but have networking

File: etc/init.d/lncm [hunk 9]
```diff
@@ -144,7 +144,7 @@ start() {
   create_swap() {
     echo "Create and install swap file"
     /bin/dd if=/dev/zero of=/media/sd/var/cache/swap bs=1M count=1024 && \
```
[the remark this answers is not in the document]
Comment (nolim1t) [reply etc/init.d/lncm]: Thats from an existing script.

File: etc/init.d/lncm [hunk 10]
```diff
@@ -190,7 +190,7 @@ start() {
     setup_user
     install_boot
     mark_sd
-    #create_swap # disable sd card swap
+    create_swap
```
[the remark this answers is not in the document]
Comment (nolim1t) [reply etc/init.d/lncm]: Maybe an option is better?

[the remark this answers is not in the document]
Comment (meeDamian) [reply etc/init.d/lncm]: I agree, see: https://github.com/lncm/pi-factory/pull/238#discussion_r331792573

File: etc/init.d/lncm-post (hunk 4 above)
Comment (nolim1t) [reply etc/init.d/lncm-post]: Good idea might actually leave it out as its not even necessary anymore

File: etc/apk/world [hunk 11]
```diff
@@ -1,7 +1,11 @@
 alpine-base
-avahi
 dbus
+avahi
+chrony
```
[the remark this answers is not in the document]
Comment (nolim1t) [reply etc/apk/world]: generated by ```setup-alpine``` when I created that image. Its for ntp

File: etc/ssh/sshd_config (hunk 5 above)
Comment (nolim1t) [reply etc/ssh/sshd_config]: done

File: etc/wpa_supplicant/wpa_supplicant.conf (hunk 6 above)
Comment (nolim1t) [reply etc/wpa_supplicant/wpa_supplicant.conf]: Its not necessary on alpine

**Comment by meeDamian** [comment]
Please revert https://github.com/lncm/pi-factory/pull/238/commits/72e829552ba663f5851921a7fcd219969ad21c8f it's actually very useful:

1. It prevents shit keys from being generated, and
1. Thanks to that speeds up initial boot time

File: etc/init.d/lncm-post (hunk 4 above)
Comment (nolim1t) [reply etc/init.d/lncm-post]: Removed

File: etc/apk/world (hunk 7 above)
[the remark this answers is not in the document]
Comment (nolim1t) [reply etc/apk/world]: sure

File: etc/init.d/lncm-online (hunk 2 above)
Comment (nolim1t) [reply etc/init.d/lncm-online]: sounds good but not really keen on touching stuff (and potentially breaking) thats already working. Just want to get a version of pi-factory that doesn't have any of that node stuff first

File: etc/init.d/lncm-post (hunk 3 above)
Comment (nolim1t) [reply etc/init.d/lncm-post]: good idea

**Comment by nolim1t** [comment]
Also using github actions to build the image too

**Comment by nolim1t** [comment]
All thats remaining now is pi zero ```armhf``` config and we are done with this PR

**Comment by nolim1t** [comment]
```./make_img.sh``` will generate two 256 MB images now.

**Comment by nolim1t** [comment]
With lack of iotwifi, this can be handled in documentation to create a wpa_supplicant file and place it in the fat partition where it will now be picked up (like raspbian does)

**Pull request merged**